### DISCUSSION TITLE
Use LitServe instead of FastAPI (enhancement)

### DISCUSSION THREAD

**Discussion by Youho99** [lhenault/simpleAI#60] [Ideas]
I just came across this [LitServe](https://github.com/Lightning-AI/LitServe) repository, which promises 2x performance over FastAPI.

LitServe is designed by the creators of pytorch-lightning, and is specifically designed for deploying AI models.

It might be interesting to test this to compare performance for SimpleAI.

**Comment by lhenault** [comment]
Hello! Thanks for flagging this project I wasn't aware of. It sounds interesting and I'll try it soon.

I think it makes sense to keep an eye on alternatives to FastAPI, but:

- Looking at the benchmark I'm not really convinced by their claim: the only relevant [graph](https://camo.githubusercontent.com/cb4734d63fc85a5d77ca449757caa6a0688d11770edb1ab5e3dc2808e5988702/68747470733a2f2f706c2d626f6c74732d646f632d696d616765732e73332e75732d656173742d322e616d617a6f6e6177732e636f6d2f6170702d322f6c735f6368617274735f76362e706e67) is the second one, where FastAPI has 6 requests per second versus 8 for LitServe on CPU (the "optimized" version doesn't seem fair as they should do the same with FastAPI then), and 45 vs 57 with a GPU. So it's more about 50% speed up in a benchmark with vested interests and no idea of statistical significance or other nice things. So I'm a bit cautious about this (although it's promising and a decent increase of performance potentially).
- For LLM use cases the underlying framework is rarely the bottleneck
- As you've been experiencing it yourself, there is more than raw speed to consider and the FastAPI ecosystem is convenient and reliable, for things like logging, CORS, middlewares, community, etc... There is a reason why FastAPI is a benchmark that other projects try to challenge regularly.

Cheers!

**Comment by Youho99** [comment]
I mentioned this project in fact more to know if for SimpleAI, it could be an advantage or not.

From what you mention, FastAPi seems to remain the best solution, in the absence of significant performance gains.

Maybe when the library has been tested by the community, its use will become clearer and indispensable.

**Comment by Youho99** [comment]
You can move this issue as a discussion

**Reply by lhenault** [reply]
Good idea and done. :)
